src/shared/nested-dropdown-list/nested-dropdown-list.tsx

```diff
@@ -112,18 +148,81 @@ export const NestedDropdownList = <V1, V2, V3>({
         onSelectItem(item);
     };
 
-    const handleExpand = () => {
+    const handleExpand = (keyPath: string[]) => {
+        const level = keyPath.length;
+
+        const lists = produce(currentItems, (draft) => {
+            let item = null;
+
+            switch (level) {
+                case 1:
+                    item = draft.get(keyPath[0]);
+                    break;
+                case 2:
+                    item = draft.get(keyPath[0]).subItems.get(keyPath[1]);
+                    break;
```

Comment:
extract this into a helper function

Reply:
i do some update for this helper, it would be share will expanded all and expanded each item. please check. it may verbose

Reply:
I was envisioning a helper for the main function to get the item from the draft state then update it:

```
produce(currentItems, (draft) => {
   const item = getItemAtKeyPath(draft, keyPath)
   item.expanded = true;
})
```

and `getItemAtKeyPath` can be simplified this way

Reply:
let me think, update this weekend or Monday. it is the loop..